src/neo/SmartContract/ApplicationEngine.Runtime.cs

```diff
@@ -293,7 +293,15 @@ protected internal BigInteger GetRandom()
         protected internal void RuntimeLog(byte[] state)
         {
             if (state.Length > MaxNotificationSize) throw new ArgumentException(null, nameof(state));
+
             string message = Utility.StrictUTF8.GetString(state);
+
+            if (!CurrentContext.GetState<ExecutionContextState>().CallFlags.HasFlag(CallFlags.AllowNotify))
+            {
+                OnFault(new Exception(message));
```

Comment:
Why are we faulting on log if AllowNotify is enabled? Why would we assume that a call to Log must be a fault?

Reply:
Because Log require Notify rights so if we don't have it, it will fault in the previous version, now the same but at least we have the message captured. That's my favorite version for solve the issue

Reply:
> Why are we faulting on log if AllowNotify is enabled? Why would we assume that a call to Log must be a fault?

If **NOT**

Reply:
OK, I see the logic. Since this PR changes `requiredCallFlags` to None, RuntimeLog will only go down the fault path in cases where we would have thrown in `ValidateCallFlags` previously.

Why are we calling OnFault instead of throwing like `ValidateCallFlags` does?

Reply:
Because with OnAult we can set the Log error message, and we don't lose the assert message